bumble/device.py

```diff
@@ -940,6 +940,11 @@ async def start_advertising(
         if self.advertising:
             await self.stop_advertising()
 
+        # add each Service's advertising data
+        for attribute in self.gatt_server.attributes:
```

Comment:
it isn't safe to mutate `self.advertisting_data` here. First, this would keep appending every time you call `start_advertisting`. Second, an app might not want to advertise the service data for all its services. That should be a choice of the code that sets `.advertising_data`. So maybe a better way it to create a helper method on the GATT server that can return a list of all the advertising data for all the registered services. The caller would then be able to call something like: `device.advertising_data = some_advertising_data + b''.join(device.get_advertising_service_data())` where (`get_advertising_service_data` would return a (possibly empty) list of advertising data elements, one per service).

Reply:
ack